### PR TITLE
TypeScript Port & GatsbyJS Update

### PR DESCRIPTION
**Pull request by Empty2k12**
This PR updates the codebase to use TypeScript and updates GatsbyJS from version `1.9.277` to `2.3.32` in preparation of the redesign.

### CODE REVIEW

File: src/templates/content.tsx [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+interface ContentProps {
+    data: any
+}
+
+class ContentTemplate extends React.Component<ContentProps> {
+    public render() {
+        const contentData = this.props.data.contentfulContent
+        const {
+            title,
+            // slug,
```
Comment (Empty2k12) [inline src/templates/content.tsx]: These are marked as comments in preparation for the redesign, where they will most likely be used. Unused ones should be commented out.

File: src/pages/index.tsx [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+
+    const day = date.getDate()
+    const monthIndex = date.getMonth()
+    const year = date.getFullYear()
+
+    return `${day} ${monthNames[monthIndex]} ${year}`;
+}
+
+interface BlogProps {
+    node: any
```
Comment (Empty2k12) [inline src/pages/index.tsx]: Use strict typing.

File: src/pages/index.tsx [hunk 3]
```diff
@@ -0,0 +1,99 @@
...
+                    </p>
+                    <p className="section_title-header">{node.title}</p>
+                </div>
+            </div>
+        </Link>
+    </section>
+)
+
+interface IndexProps {
+    data: any
```
Comment (Empty2k12) [inline src/pages/index.tsx]: Use strict typing.

File: src/pages/index.tsx [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+}
+
+class IndexPage extends React.Component<IndexProps> {
+    public render() {
+        const usProductEdges = this.props.data.us.edges
+        return (
+            <DefaultLayout>
+                {usProductEdges.map(({node}: {node: any}, {i}: {i: any}) => (
```
Comment (Empty2k12) [inline src/pages/index.tsx]: Use strict typing.

File: src/templates/content.tsx [hunk 5]
```diff
@@ -0,0 +1,69 @@
+import { graphql } from 'gatsby'
+import Img from 'gatsby-image'
+import React from 'react'
+
+import DefaultLayout from '../components/layout'
+
+interface ContentProps {
+    data: any
```
Comment (Empty2k12) [inline src/templates/content.tsx]: Use strict typing.

File: src/styles/index.css [hunk 6]
```diff
@@ -27,9 +29,10 @@ a:visited {
 /*
 * Header Styles
 */
+
 header {
     font-size: 8vw;
-    color: #D7BE7E;
+    color: #d7be7e;
```
Comment (msrd0) [inline src/styles/index.css]: dafür auch

File: src/styles/index.css [hunk 7]
```diff
@@ -1,14 +1,16 @@
 /*
 * General purpose / helper Styles
 */
+
 body {
     padding: 0;
     margin: 0;
-    background:#0F2C42;
+    background: #0f2c42;
```
Comment (msrd0) [inline src/styles/index.css]: dafür benutzt man bitteschön variablen

File: src/styles/index.css [hunk 8]
```diff
@@ -110,13 +114,16 @@ nav {
     font-size: 1.5vw;
 }
 
-.nav-button:hover, .nav-search:hover, .box:hover {
-    box-shadow: 0.5vw 0.5vw 0 #D7BE7E;
+.nav-button:hover,
+.nav-search:hover,
+.box:hover {
+    box-shadow: 0.5vw 0.5vw 0 #d7be7e;
```
Comment (msrd0) [inline src/styles/index.css]: ich wiederhole mich

File: src/styles/index.css [hunk 9]
```diff
@@ -140,7 +147,7 @@ footer {
 
 .footer-links {
     display: flex;
-    border-top: 0.2vw solid #D7BE7E;
+    border-top: 0.2vw solid #d7be7e;
```
Comment (msrd0) [inline src/styles/index.css]: und nochmal

File: src/templates/content.tsx [hunk 10]
```diff
@@ -0,0 +1,69 @@
...
+                </article>
+            </DefaultLayout>
+        )
+    }
+}
+
+export default ContentTemplate
+
+export const pageQuery = graphql`
+    query contentQuery($id: String!) {
```
Comment (msrd0) [inline src/templates/content.tsx]: benutz dafür typesafe builders und nicht sonen string müll

File: tslint.json [hunk 11]
```diff
@@ -0,0 +1,30 @@
...
+            false
+        ],
+        "variable-name": [
+            true,
+            "ban-keywords",
+            "check-format",
+            "allow-pascal-case"
+        ]
+    }
+}
```
Comment (msrd0) [inline tslint.json]: no trailing newline. igitt

File: src/pages/index.tsx [hunk 12]
```diff
@@ -0,0 +1,99 @@
+import { graphql } from 'gatsby'
+import Img from 'gatsby-image'
+import Link from 'gatsby-link'
+import React from 'react'
+
+import DefaultLayout from '../components/layout'
+
+import '../styles/index.css'
+
+function formatDate(date: Date) {
```
[the remark this answers is not in the document]
Comment (Empty2k12) [reply src/pages/index.tsx]: Nein, aber kommt eh bald weg.

**Comment by msrd0** [comment]
I'd have done some more changes before merging into master

**Pull request merged**